temporalcli/commands.workflow.go

```diff
@@ -88,6 +92,117 @@ func (c *TemporalWorkflowDeleteCommand) run(cctx *CommandContext, args []string)
 	return nil
 }
 
+func (c *TemporalWorkflowUpdateOptionsCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	if c.VersioningOverrideBehavior.Value == "unspecified" || c.VersioningOverrideBehavior.Value == "auto_upgrade" {
+		if c.VersioningOverrideSeriesName != "" {
+			return fmt.Errorf("cannot set deployment series name with %v behavior", c.VersioningOverrideBehavior)
+		}
+		if c.VersioningOverrideBuildId != "" {
+			return fmt.Errorf("cannot set deployment build ID with %v behavior", c.VersioningOverrideBehavior)
+		}
+	}
+
+	if c.VersioningOverrideBehavior.Value == "pinned" {
+		if c.VersioningOverrideSeriesName == "" {
+			return fmt.Errorf("missing deployment series name with 'pinned' behavior")
+		}
+		if c.VersioningOverrideBuildId == "" {
+			return fmt.Errorf("missing deployment build ID with 'pinned' behavior")
+		}
+	}
+
+	exec, batchReq, err := c.workflowExecOrBatch(cctx, c.Parent.Namespace, cl, singleOrBatchOverrides{})
+
+	// Run single or batch
+	if err != nil {
+		return err
+	} else if exec != nil {
+		behavior := workflow.VersioningBehaviorUnspecified
+		switch c.VersioningOverrideBehavior.Value {
+		case "unspecified":
+		case "pinned":
+			behavior = workflow.VersioningBehaviorPinned
+		case "auto_upgrade":
+			behavior = workflow.VersioningBehaviorAutoUpgrade
+		default:
+			return fmt.Errorf(
+				"invalid deployment behavior: %v, valid values are: 'unspecified', 'pinned', and 'auto_upgrade'",
+				c.VersioningOverrideBehavior,
+			)
+		}
+
+		_, err := cl.UpdateWorkflowExecutionOptions(cctx, client.UpdateWorkflowExecutionOptionsRequest{
+			WorkflowId: exec.WorkflowId,
+			RunId:      exec.RunId,
+			WorkflowExecutionOptionsChanges: client.WorkflowExecutionOptionsChanges{
+				VersioningOverride: &client.VersioningOverride{
+					Behavior: behavior,
+					Deployment: client.Deployment{
+						SeriesName: c.VersioningOverrideSeriesName,
+						BuildID:    c.VersioningOverrideBuildId,
+					},
+				},
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to update workflow options: %w", err)
+		}
+		cctx.Printer.Println("Update workflow options succeeded")
+	} else { // Run batch
+		var workflowExecutionOptions *workflowpb.WorkflowExecutionOptions
+		protoMask, err := fieldmaskpb.New(workflowExecutionOptions, "versioning_override")
+		if err != nil {
+			panic("invalid field mask")
```

Comment:
While I know this thing shouldn't happen, I think we can still use regular errors here